### PR TITLE
Fix angle attr for Label annotation

### LINKED ISSUE #4278: Label doesn't accept angle property
The following code adds the label, but doesn't rotate it.

```python
misc = Label(x=1, y=1, text=["Some stuff"], angle=np.pi/3)
plot.add_layout(misc)
```
Both the python and bokehjs side have an angle property and the bokehjs seems to attempt to use it (https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/models/annotations/label.coffee#L105)

### PR DESCRIPTION
**Pull request by canavandl**
- [x] issues: fixes #4278 
- [x] tests added / passed

Now:

screen shot 2016-05-04 at 6 23 00 pm

There is one thing to note (and this is mentioned in the Label `angle` attr docstring):

>The center of rotation for canvas and css render_modes is different. For `render_mode="canvas"` the label is rotated from the top-left corner of the annotation, while for `render_mode="css"` the annotation is rotated around it's center.

Getting rotated label locations to match for css and canvas modes seems very difficult and not worth resolving, so I'm ok with adding a warning.

### CODE REVIEW

**Comment by canavandl** [comment]
ping @birdsarah

**Comment by bryevdv** [comment]
FYI the merge conflict is in the screenshot:
```
CONFLICT (content): Merge conflict in tests/integration/annotations/screenshots/base__test_label__test_label.png
```

**Comment by birdsarah** [comment]
Rockin!

I'll check out the code in a moment.

For the merge conflict, I added new screenshots to @mattpap's TS API PR. So you just need to add a new screenshot.

He made a good point which is that we should have toolbars on tests if we don't need them - these are meant to be highly isolated tests that only test the specific feature. I definitely agree. Screenshot tests are horrible for false negatives so we have to be really careful. 

So, although it's a little out-of-scope, I think we should update the integration test in that vain:
1) don't use the plotting api - use Plot, 
2) use Range1d (it's more specific and has less auto-magic)
3) use toolbar_location=None on Plot.

File: bokehjs/src/coffee/models/annotations/label.coffee [hunk 1]
```diff
@@ -143,6 +144,7 @@ class LabelView extends Renderer.View
         'font-size': "#{@_text_font_size[i]}"
         'font-family': "#{@mget('text_font')}"
         'background-color': "#{@visuals.background_fill.color_value()}"
+        'transform': "rotate(#{rad_to_deg(@_angle[i])}deg)"
```
Comment (birdsarah) [inline bokehjs/src/coffee/models/annotations/label.coffee]: I'm a little confused - is angle always in radians?

I've been hunting around and I think the answer is yes. You can provide it in degrees python side (via AngleSpec), but then it comes down the pipe in radians? 

I just want to confirm this.

File: bokehjs/src/coffee/models/annotations/label.coffee [hunk 2]
```diff
@@ -102,8 +103,8 @@ class LabelView extends Renderer.View
     for i in [0...@_text.length]
       ctx.save()
 
-      ctx.rotate(@mget('angle'))
       ctx.translate(@sx[i] + @_x_offset[i], @sy[i] - @_y_offset[i])
+      ctx.rotate(@_angle[i])
```
Comment (birdsarah) [inline bokehjs/src/coffee/models/annotations/label.coffee]: Definitely good to move this after the translate. I'm pretty sure I'll be using translate for my side panels work, so this makes a lot more sense.

**Comment by birdsarah** [comment]
Code looks good. A few questions and mostly a note about the screenshot test.

File: bokehjs/src/coffee/models/annotations/label.coffee (hunk 1 above)
Comment (mattpap) [inline bokehjs/src/coffee/models/annotations/label.coffee]: > I'm a little confused - is angle always in radians?

See https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/core/properties.coffee#L224

Comment (mattpap) [inline bokehjs/src/coffee/models/annotations/label.coffee]: I'm not sure why convert to degrees here. `rotate()` transform supports radians and other (see https://developer.mozilla.org/en-US/docs/Web/CSS/angle).

Comment (canavandl) [inline bokehjs/src/coffee/models/annotations/label.coffee]: I missed that when I read the spec. Thanks for pointing that out @mattpap

**Comment by canavandl** [comment]
superceded by #4297

**Pull request closed without merging**